balancer_v1_wrapper.go

```diff
@@ -315,12 +313,12 @@ func (bw *balancerWrapper) Pick(ctx context.Context, opts balancer.PickOptions)
 			Metadata:   a.Metadata,
 		}]
 		if !ok && failfast {
```

Comment:
@menghanl this is a bug, right?  We should return an error regardless of `failfast`, shouldn't we?  Otherwise, we'll be trying to use `sc` (which will be `nil`) in the lookup below, and then eventually returning it from the function with no error.

Reply:
Returning `nil` will not cause error because we [check it](https://github.com/grpc/grpc-go/blob/187e357ac194b625331eb5f35c6440c40770f018/picker_wrapper.go#L153). `nil` will result in a re-pick.

We will only get `nil` here when things are not in sync (between the wrapper and the v1-balancer). This means the connectivity state has changed, and a new pick was (or will be) updated. So I think re-pick makes sense.

Reply:
But we will also get a re-pick if we return the transient failure error immediately.  Technically this works as-is, but if an error happens we should be returning an error, not a nil sc with no error.

Returning a nil sc will also result in the info log (which should probably be a warning) "subconn returned from pick is not *acBalancerWrapper", which shouldn't normally be happening (and it puts our implementation details into the user's log messages).

OK, let's go ahead with this PR but I think we should change this separately.  I'll send a PR.